CMakeLists.txt

```diff
@@ -67,84 +74,93 @@ if(ASIO_CPM_FETCHCONTENT_COMPAT OR NOT ASIO_USE_CPM)
                        "${ASIO_VERSION}"
   )
-
   set(ASIO_VERSION
       ${ASIO_VERSION}
-      CACHE STRING "asio library version" FORCE
+      CACHE STRING "Asio library version" FORCE
   )
 
-  # Configure ##
-
-  add_library(asio STATIC src/asio.cpp)
+  # Library
+  add_library(asio INTERFACE)
   add_library(asio::asio ALIAS asio)
   target_include_directories(
-    asio PUBLIC $<BUILD_INTERFACE:${asio_SOURCE_DIR}/asio/include>
-                $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
+    asio SYSTEM INTERFACE $<BUILD_INTERFACE:${asio_SOURCE_DIR}/asio/include>
+                          $<INSTALL_INTERFACE:${CMAKE_INSTALL_INCLUDEDIR}>
   )
 
-  find_package(Threads)
-  target_link_libraries(asio PRIVATE Threads::Threads)
+  find_package(Threads REQUIRED)
+  target_link_libraries(asio INTERFACE Threads::Threads)
 
-  target_compile_features(asio PUBLIC cxx_std_11)
+  set_target_properties(
+    asio
+    PROPERTIES CXX_STANDARD 11
+               CXX_STANDARD_REQUIRED ON
+               CXX_EXTENSIONS OFF
+  )
   target_compile_definitions(
-    asio PUBLIC -DASIO_STANDALONE -DASIO_SEPARATE_COMPILATION -DASIO_NO_DEPRECATED
+    asio INTERFACE ASIO_STANDALONE $<$<BOOL:${ASIO_NO_DEPRECATED}>:ASIO_NO_DEPRECATED>
   )
 
-  # Fix warning : "Please define _WIN32_WINNT or _WIN32_WINDOWS appropriately."
+  # Fix warning: "Please define _WIN32_WINNT or _WIN32_WINDOWS appropriately."
   # https://stackoverflow.com/questions/9742003/platform-detection-in-cmake
   if(WIN32 AND CMAKE_SYSTEM_VERSION)
     set(ver ${CMAKE_SYSTEM_VERSION})
     string(REPLACE "." "" ver ${ver})
     string(REGEX REPLACE "([0-9])" "0\\1" ver ${ver})
     set(version "0x${ver}")
-    target_compile_definitions(asio PUBLIC -D_WIN32_WINNT=${version})
+    target_compile_definitions(asio INTERFACE _WIN32_WINNT=${version})
   endif()
-
 endif()
 
+#
+# Examples
+#
 if(ASIO_ENABLE_EXAMPLES)
   enable_testing()
   add_executable(
-    asio_example_cpp03_timers_time_t_timer
-    ${asio_SOURCE_DIR}/asio/src/examples/cpp03/timers/time_t_timer.cpp
+    asio_example_cpp11_timers_time_t_timer
+    ${asio_SOURCE_DIR}/asio/src/examples/cpp11/timers/time_t_timer.cpp
   )
-  target_link_libraries(asio_example_cpp03_timers_time_t_timer PRIVATE asio::asio)
-  add_custom_target(asio_examples DEPENDS asio_example_cpp03_timers_time_t_timer)
-  add_test(NAME asio_example COMMAND asio_example_cpp03_timers_time_t_timer)
-
+  target_link_libraries(asio_example_cpp11_timers_time_t_timer PRIVATE asio::asio)
+  add_custom_target(asio_examples DEPENDS asio_example_cpp11_timers_time_t_timer)
+  add_test(NAME asio_example COMMAND asio_example_cpp11_timers_time_t_timer)
 endif()
 
+#
+# Install
+#
 if(ASIO_ENABLE_INSTALL)
-  include(cmake/get_cpm.cmake)
-  CPMAddPackage("gh:OlivierLDff/PackageProject.cmake@1.9.0")
+  if(NOT CPM_INITIALIZED)
+    include(cmake/get_cpm.cmake)
+  endif()
+  CPMAddPackage("gh:TheLartians/PackageProject.cmake@1.12.0")
   packageProject(
```

Comment:
the header only installation are independent of architecture and should not installed to <prefix>/lib

Reply:
Can you check PR https://github.com/OlivierLDff/asio.cmake/pull/35 and add a comment there.
Thanks 🙏